__tests__/test-data.test.js

```diff
@@ -0,0 +1,166 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("api/topics", () => {
+  test("returns an array ", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((body) => {
+        const arrayOfTopics = body.body.topics;
+        expect(Array.isArray(arrayOfTopics)).toBe(true);
+      });
+  });
+  test("expect 200 status code , check if each obj in array has the keys slug and description  ", () => {
+    return request(app)
+      .get("/api/topics")
+
+      .then((body) => {
+        const arrayOfTopics = body.body.topics;
+
+        const areKeysPresent = arrayOfTopics.every((obj) => {
+          if (obj.slug !== null && obj.description !== null) {
+            return true;
+          } else {
+            return false;
+          }
+        });
+        expect(areKeysPresent).toBe(true);
+      });
+  });
+
+  test("returns 404 error if topics is not present after /api ", () => {
+    return (
+      request(app)
+        .get("/api/topic")
+        // wrong end point name
+        .expect(404)
+    );
+  });
+});
+
+
+
+
+
+
+
+
+
+describe("get /api", () => {
+  test("200,returns an object ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        expect(Array.isArray(endPointData)).toBe(false);
+        expect(typeof endPointData).toBe("object");
+      });
+  });
+  test("200, check if the key in each object matches the api path and contains information  ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const arraKey = Object.keys(endPointData);
+        expect(arraKey).toEqual([
+          "GET /api",
+          "GET /api/topics",
+          "GET /api/articles",
+        ]);
+      });
+  });
+  test("200, check if each nested obj contains a description with a information inisde it  ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const endPointKeys = Object.keys(endPointData);
+        const everykeyHasDescriptionInfo = endPointKeys.every((key) => {
+          if (endPointData[key].description !== null) {
```

Comment:
I see what you're trying to do but you can't just check !== null as it will never be null, it would be better to use toHaveProperty